analysis/study_definition_ethnicity.py

```diff
@@ -16,45 +16,99 @@
     },
     index_date=end_date,
     population=patients.all(),
-    ethnicity=patients.categorised_as(
+    eth=patients.with_these_clinical_events(
+        ethnicity16_codes,
+        returning="category",
+        find_last_match_in_period=True,
+        include_date_of_match=False,
+        return_expectations={
+            "category": {
+                "ratios": {
+                    "1": 0.1,
+                    "2": 0.1,
+                    "3": 0.1,
+                    "4": 0.1,
+                    "5": 0.1,
+                    "6": 0.025,
+                    "7": 0.025,
+                    "8": 0.05,
+                    "9": 0.05,
+                    "10": 0.05,
+                    "11": 0.05,
+                    "12": 0.05,
+                    "13": 0.05,
+                    "14": 0.05,
+                    "15": 0.05,
+                    "16": 0.05,
+                }
+            },
+            "incidence": 0.75,
+        },
+    ),
+    ethnicity16=patients.categorised_as(
+        {
+            "Unknown": "DEFAULT",
+            "White - British": "eth='1'",
+            "White - Irish": "eth='2'",
+            "White - Any other White background": "eth='3'",
+            "Mixed - White and Black Caribbean": "eth='4'",
+            "Mixed - White and Black African": "eth='5'",
+            "Mixed - White and Asian": "eth='6'",
+            "Mixed - Any other mixed background": "eth='7'",
+            "Asian or Asian British - Indian": "eth='8'",
+            "Asian or Asian British - Pakistani": "eth='9'",
+            "Asian or Asian British - Bangladeshi": "eth='10'",
+            "Asian or Asian British - Any other Asian background": "eth='11'",
+            "Black or Black British - Caribbean": "eth='12'",
+            "Black or Black British - African": "eth='13'",
+            "Black or Black British - Any other Black background": "eth='14'",
+            "Other Ethnic Groups - Chinese": "eth='15'",
+            "Other Ethnic Groups - Any other ethnic group": "eth='16'",
+        },
```

Comment:
The codelist also has "17 - Not stated"
Just checking it's a choice to include that as Unknown?
I extracted them separately but then ended up combining them for my analysis
But extracting them together protects against any potential small number suppression issues

Reply:
oh yes, 17 is in the about but there's no code, so I assume its just missing? [^0] and I couldnt find code for "17' in your code? [^1]

[^0]:  https://www.opencodelists.org/codelist/opensafely/ethnicity-snomed-0removed/2e641f61/
[^1]: https://github.com/opensafely/antidepressant-prescribing-lda/blob/12e4acca89a73c640dc14c602ef65c4c6640894b/analysis/study_definition_ethnicity.py

Reply:
Ah good point that there is no code 17 in the csv. I'll ask Colm about that